src/Zio/UPathExtensions.cs

```diff
@@ -239,28 +239,26 @@ public static bool IsInDirectory(this UPath path, UPath directory, bool recursiv
     /// </summary>
     /// <param name="path">The path.</param>
     /// <param name="name">The name of a parameter to include n the <see cref="ArgumentNullException"/>.</param>
-    /// <returns>A path not modified.</returns>
     /// <exception cref="System.ArgumentNullException">If the path was null using the parameter name from <paramref name="name"/></exception>
-    public static UPath AssertNotNull(this UPath path, string name = "path")
+    public static void AssertNotNull(this UPath path, string name = "path")
     {
-        if (path.FullName is null)
-            throw new ArgumentNullException(name);
-        return path;
+        if (path.IsNull)
```

Comment:
`FullName is null` -> `IsNull` and the local method to move out the exception to help inline it.